sql/src/test/java/ddlog/RuntimeTest.java

```diff
@@ -0,0 +1,1064 @@
+        Assert.assertEquals("CREATE VIEW v1 AS\n" +
+                "SELECT *\n" +
+                "FROM\n" +
+                "  t1\n" +
+                "WHERE (column1 = 10)\n", stat);
+    }
+
+    @Test
+    public void testExpression() {
+        Translator t = new Translator(null);
+        DDlogIRNode node = t.translateExpression("true");
+        String s = node.toString();
+        Assert.assertEquals("true", s);
+
+        node = t.translateExpression("'some string'");
+        s = node.toString();
+        Assert.assertEquals("\"some string\"", s);
+
+        node = t.translateExpression("((1 + 2) >= 0) or ((1 - 3) <= 0)");
+        s = node.toString();
+        Assert.assertEquals("(((64'sd1 + 64'sd2) >= 64'sd0) or ((64'sd1 - 64'sd3) <= 64'sd0))", s);
+
+        node = t.translateExpression("CASE 0\n" +
+                "      WHEN 1 THEN 'One'\n" +
+                "      WHEN 2 THEN 'Two'\n" +
+                "      ELSE 'Other'\n" +
+                "END");
+        s = node.toString();
+        Assert.assertEquals("if (64'sd0 == 64'sd1){\n" +
+                "\"One\"} else {\n" +
+                "if (64'sd0 == 64'sd2){\n" +
+                "\"Two\"} else {\n" +
+                "\"Other\"}}", s);
+    }
+
+    @Test
+    public void testWhen() {
+        String query = "create view v1 as SELECT DISTINCT CASE WHEN column1 = 1 THEN 1 WHEN 1 < column1 THEN 2 ELSE 3 END FROM t1";
+        String program =
+                this.header(false) +
+                        "typedef Ttmp0 = Ttmp0{col1:signed<64>}\n" +
+                        this.relations(false) +
+                        "relation Rtmp0[Ttmp0]\n" +
+                        "output relation Rv1[Ttmp0]\n" +
+                        "Rv1[v3] :- Rt1[v0],var v2 = Ttmp0{.col1 = if (v0.column1 == 64'sd1){\n" +
+                        "64'sd1} else {\n" +
+                        "if (64'sd1 < v0.column1){\n" +
+                        "64'sd2} else {\n" +
+                        "64'sd3}}},var v3 = v2.";
+        testTranslation(query, program);
+    }
+
+    @Test
+    public void testWhenWNull() {
+        String query = "create view v1 as SELECT DISTINCT CASE WHEN column1 = 1 THEN 1 WHEN 1 < column1 THEN 2 ELSE 3 END FROM t1";
+        String program =
+            this.header(true) +
+                "typedef Ttmp0 = Ttmp0{col1:signed<64>}\n" +
+                this.relations(true) +
+                "relation Rtmp0[Ttmp0]\n" +
+                "output relation Rv1[Ttmp0]\n" +
+                "Rv1[v3] :- Rt1[v0],var v2 = Ttmp0{.col1 = if unwrapBool(a_eq_NR(v0.column1, 64'sd1)){\n" +
+                "64'sd1} else {\n" +
+                "if unwrapBool(a_lt_RN(64'sd1, v0.column1)){\n" +
+                "64'sd2} else {\n" +
+                "64'sd3}}},var v3 = v2.";
+        testTranslation(query, program, true);
+    }
+
+    @Test
+    public void testAlias() {
+        String query = "create view v1 as SELECT DISTINCT t2.column1 FROM t1 AS t2";
```

Comment:
It looks the alias is ignored in the ddlog program because it is not required in this instance. That wouldn't be the case with things like select expressions right (they'd affect the names in the output relation)?

Reply:
Ah I see them being tested elsewhere in testMultiJoin().